src/spec/test/metaprogramming/InterceptionThroughMetaClassTest.groovy

```diff
@@ -0,0 +1,23 @@
+package metaprogramming
+
+import groovy.xml.Entity
+
+// tag::meta_class_interception[]
+class InterceptionThroughMetaClassTest extends GroovyTestCase {
+    void testPOJOMetaClassInterception() {
+        String invoking = 'ha'
+        invoking.metaClass.invokeMethod = {String name, Object args ->
+            'invoked'
+        }
+        assertEquals invoking.length(), invoking.someMethod()
```

Comment:
I'm no expert on metaprogramming, but I'm surprised that this test seems to work on Jdk8. I would have expected it to fail like this http://ci.groovy-lang.org/viewLog.html?buildId=16276&buildTypeId=Groovy_Jdk7Build every time. 

I suggest using `assert invoking.someMethod() == 'invoked'` instead. Imho this makes the intent a bit clearer.